### DISCUSSION TITLE
Sonoff Dual R3 - 2 external toggles

### DISCUSSION THREAD

**Discussion by dribeiro09** [RavenSystem/esp-homekit-devices#2103] [MEPLHAA Scripts]
Hi,

I have a Sonoff Dual R3 which I have flashed and configured using the following MELPHAA script (from here https://github.com/RavenSystem/esp-homekit-devices/discussions/1993):

`{"c":{"io":[[[13],3],[[0,32,33],6],[[14,27],2]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":27}]},"1":{"r":[{"g":27,"v":1}]},"b":[{"g":0,"t":1},{"g":32,"t":0}],"s":5},{"0":{"r":[{"g":14}]},"1":{"r":[{"g":14,"v":1}]},"b":[{"g":33,"t":0}],"s":5}]}`

This allows me to turn on and off 2 lamps which are connected to two toggle switches. However, the behaviour I intended is still not there as for me to turn on and off the lamp I have to perform two pressing cycles. Example:

- Light is off and toggle is in the unpressed position: I set the toggle to the press position (press it) -> Light turns on. If I want to turn the light off I wanted it to only be necessary to set the toggle to the unpressed position. However, I need to set it to unpressed and set it to pressed again for the light to turn off.

The behaviour previously described is the same for both lights/swiches. Is there a way for me to change this script to match the desired configuration. I've tried a couple variations but it seems to make no difference. Im certainly changing the wrong parameters. 

Appreciate all the help I can get. Keep up with the good work.

**Comment by XMP40** [comment]
Hello,
I have exact same problem with the json. Could you solve it in meantime? I also would like to connect two switches, which turn on and off respective relay by switching in external switch on and off.
Hope you can help me and maybe you can provide me a MELPHAA script?
Thanks a lot and best regfards
Mario

**Comment by Tellicious** [comment]
See this in the Wiki: [Button Press Type](https://github.com/RavenSystem/esp-homekit-devices/wiki/accessory-configuration#press-type-example)

Try this:
`{"c":{"io":[[[13],3],[[0,32,33],6],[[14,27],2]],"l":13,"b":[[0,5]]},"a":[{"0":{"r":[[27]]},"1":{"r":[[27,1]]},"b":[[0],[32],[32,0]],"s":5},{"0":{"r":[[14]]},"1":{"r":[[14,1]]},"b":[[33],[33,0]],"s":5}]}`

**Reply by XMP40** [reply]
Thanks a lot for your help.
Have a good day.
